Terminal.Gui/View/View.Navigation.cs

```diff
@@ -523,11 +520,17 @@ public bool SetFocus ()
     /// <exception cref="InvalidOperationException"></exception>
     private (bool focusSet, bool cancelled) SetHasFocusTrue (View? currentFocusedView, bool traversingUp = false)
     {
-        Debug.Assert (SuperView is null || View.IsInHierarchy (SuperView, this));
+        Debug.Assert (SuperView is null || IsInHierarchy (SuperView, this));
 
         // Pre-conditions
         if (_hasFocus)
         {
+            //// See https://github.com/gui-cs/Terminal.Gui/pull/4013#issuecomment-2823934197
+            //if (Application.Navigation is { } && (Application.Navigation.GetFocused () == this || Application.Navigation.GetFocused () == MostFocused))
+            //{
+            //    throw new InvalidOperationException (@"Do not SetFocus on a view that is already MostFocused.");
+            //}
```

Comment:
If this break tests is because there are views that when are closed doesn't set the `MostFocused` view by not calling the `Application.Navigation?.SetFocused` method.

Reply:
I am not going to fix that here. I'll look at it in

- #4013